syntax = "proto3";

message QueryRequest {
  repeated float vector = 1;
  int32 k = 2;
}

message QueryResponse {
  repeated int32 labels = 1;
  repeated float distances = 2;
}

message SetEfRequest {
  int32 ef_search = 1;
}

message SetEfResponse {
  string status = 1;
  int32 new_ef_search = 2;
}

message VectorData {
  int64 id = 1;
  repeated float vector = 2;
}

message UpdateIndexRequest {
  repeated VectorData vectors = 1;
}

